src/framework/components/rigid-body/component.js

```diff
@@ -72,99 +72,263 @@ class RigidBodyComponent extends Component {
+                this.disableSimulation();
+                this.enableSimulation();
+            }
+        }
+    }
+
+    get linearDamping() {
+        return this._linearDamping;
+    }
+
+    set linearDamping(damping) {
+        if (this._linearDamping !== damping) {
+            this._linearDamping = damping;
+
+            if (this._body) {
+                this._body.setDamping(damping, this._angularDamping);
+            }
+        }
+    }
+
+    get linearFactor() {
+        return this._linearFactor;
+    }
+
+    set linearFactor(factor) {
+        if (!this._linearFactor.equals(factor)) {
+            this._linearFactor.copy(factor);
+
+            if (this._body && this._type === BODYTYPE_DYNAMIC) {
+                ammoVec1.setValue(factor.x, factor.y, factor.z);
+                this._body.setLinearFactor(ammoVec1);
+            }
+        }
+    }
 
     get linearVelocity() {
-        var body = this.body;
-        if (body && this.type === BODYTYPE_DYNAMIC) {
-            var vel = body.getLinearVelocity();
-            this._linearVelocity.set(vel.x(), vel.y(), vel.z());
+        if (this._body && this._type === BODYTYPE_DYNAMIC) {
+            const velocity = this._body.getLinearVelocity();
+            this._linearVelocity.set(velocity.x(), velocity.y(), velocity.z());
         }
         return this._linearVelocity;
     }
 
-    set linearVelocity(lv) {
-        var body = this.body;
-        if (body && this.type === BODYTYPE_DYNAMIC) {
-            body.activate();
+    set linearVelocity(velocity) {
+        if (this._body && this._type === BODYTYPE_DYNAMIC) {
+            this._body.activate();
 
-            ammoVec1.setValue(lv.x, lv.y, lv.z);
-            body.setLinearVelocity(ammoVec1);
+            ammoVec1.setValue(velocity.x, velocity.y, velocity.z);
+            this._body.setLinearVelocity(ammoVec1);
 
-            this._linearVelocity.copy(lv);
+            this._linearVelocity.copy(velocity);
         }
     }
 
-    get angularVelocity() {
-        var body = this.body;
-        if (body && this.type === BODYTYPE_DYNAMIC) {
-            var vel = body.getAngularVelocity();
-            this._angularVelocity.set(vel.x(), vel.y(), vel.z());
+    get mask() {
+        return this._mask;
+    }
+
+    set mask(mask) {
+        if (this._mask !== mask) {
+            this._mask = mask;
+
+            // re-enabling simulation adds rigidbody back into world with new masks
+            if (this.enabled && this.entity.enabled) {
+                this.disableSimulation();
+                this.enableSimulation();
+            }
         }
-        return this._angularVelocity;
     }
 
-    set angularVelocity(av) {
-        var body = this.body;
-        if (body && this.type === BODYTYPE_DYNAMIC) {
-            body.activate();
+    get mass() {
+        return this._mass;
+    }
 
-            ammoVec1.setValue(av.x, av.y, av.z);
-            body.setAngularVelocity(ammoVec1);
+    set mass(mass) {
```

Comment:
Thinking out loud, if a mass for a dynamic body would be set to zero here, then the component system would still treat it as a dynamic body, while Ammo will change it to static. Perhaps, we can update the type here as well, which would call disableSimulation, which in turn would remove the dynamic object from the system._dynamic list.

Reply:
I've wondered about stuff like that too. So, a couple of things:

1. This PR shouldn't change any existing functionality. The physics engine should behave exactly the same as before (but with better structure and performance). If we changed things so that `this.entity.rigidbody.mass = 0;` switched the body's type to static, that could potentially change the behavior of some peoples' apps.
2. I'm not a fan of implicitly changing the value of one component's property by setting another. Feels too 'magical' to me (to use a word @Maksims likes to use).

Reply:
Yes, that makes sense. This is simply the default Ammo behavior when one updates the mass of a rigidbody. Thought it was worth mentioning.
https://github.com/kripken/ammo.js/blob/99d0ec0b1e26d7ccc13e013caba8e8a5c98d953b/bullet/src/BulletDynamics/Dynamics/btRigidBody.cpp#L231-L241